google/cloud/logging_v2/entries.py

```diff
@@ -337,10 +337,11 @@ def to_api_repr(self):
         """API repr (JSON format) for entry."""
         info = super(ProtobufEntry, self).to_api_repr()
         proto_payload = None
-        if self.payload_json:
-            proto_payload = dict(self.payload_json)
-        elif self.payload_pb:
-            proto_payload = MessageToDict(self.payload_pb)
+        if self.payload is not None:
+            try:
+                proto_payload = MessageToDict(self.payload)
+            except Exception:
```

Comment:
Can you replace this bare exception with the exception sub-class you'd expect to catch here?